### DISCUSSION TITLE
Request

### DISCUSSION THREAD

**Discussion by IanKBrown** [pymedusa/Medusa#10760] [Ideas & Feature requests]
Greetings, I have a couple of requests not so much issues.

In the show list view there is a field downloads. I would appreciate it if the 3 values were defined a follows:
First number = number of episodes downloaded
Second number = number of episodes snatched
Third number = number of episodes available

I for example have as download quality parameters 720 HDTV, 1080 HDTV, 720 WEB-DL, 1080 WEB-DL with  as preferred quality 720 WEB-DL. It is possible that you get to see 21+1/22. This I find confusing as in this specific case 22 episodes have been downloaded, 1 episode has been snatched and there is a total of 22 episodes. It is however also possible that 21 episodes have been downloaded, 1 snatched and there is a total of 22 episodes. If you want to check what episodes are missing you need open the record and check what the status is.

Second request is that if and episode has been downloaded and the episode has been snatched (this I the case if the preferred quality was not available at the time of downloading) the field subtitles is blanked. Could you just indicate that an episode has been snatched and display all other information related to the downloaded episode? I check to see if all required subtitles have ben downloaded.

**Comment by p0psicles** [comment]
For no 1.

Sorry I've read you requeste like 10x now. But I still don't understand your point.
its
Total number of downloaded + total number of snatched = total episodes.
22 downloaded, 1 snatched and total of 22 doesn't make sense. That shouldn't be the case. Because the state for an ep can only be downloaded OR snatched. If it was downloaded before, and got re-snatched after. It's doesn't count as downloaded anymore.

for no 2.
I'll see what I can do.

**Reply by IanKBrown** [reply]
Hi there. Thanks for responding. I'll try to clarify.

First field: Total number of downloaded. This should show the number of files/episodes that have been downloaded. This is not the case as is demonstrated in the attached screenshot Finding Alice 01 (5+1/6). 6 files/episodes have been downloaded and 1 was snatched (I have defined that my preferred format is 720 webdl. As you can see in screenshot Finding Alice 02 6 files have been downloaded and 1 snatched.

Second field: Total number of snatched. This should reflect the number of files snatched so it is perfectly possible that the sum of downloaded + sum of snatched is greater that the number of episodes available.

Third field: I now understand that you add downloaded and snatched (this is not correct). I would find it far more informative
to have the following information:
Field 1: number of downloaded files/episodes
Field 2: number of files/episodes snatched
Field 3: number of episodes available

As I pointed out it is a pain that if a file/episode is snatched (see screenshot Finding Alice 2 & 3) the information related to the downloaded file (subtitles is blanked).

I appreciate your trouble and I hope I have made things a bit clearer.

Take care,

Ian

<img width="1792" alt="Finding Alice 01" src="https://user-images.githubusercontent.com/44754804/175304327-82fc0c52-9e0f-49dd-afa4-942f4c5f0494.png">
<img width="1792" alt="Finding Alice 02" src="https://user-images.githubusercontent.com/44754804/175304332-02d9d7a6-1bea-49a3-b847-ced4d4115c75.png">
<img width="1792" alt="Finding Alice 03" src="https://user-images.githubusercontent.com/44754804/175304338-644a8940-8fe7-4c84-9b69-ec544c344595.png">

**Comment by p0psicles** [comment]
Okay I think now I understand where your coming from.
The issue currently is that we store one state for the episode. So if an ep gets status "Downloaded", and after that you snatch it. It changes from "Downloaded" -> "Snatched". So currently we only know if a "Snatched" file already had a download, when we check the `location` field. Or when we perform additional queries on the `/api/v2/episode/history` api endpoint. But that's used when you navigate to the snatchSelection page.

I can look into our stats queries, to see if something is possible like this, without having to do a lot of data manipulation.

As for the subtitles, I think that's an easy fix. (https://github.com/pymedusa/Medusa/pull/10778)

**Reply by p0psicles** [reply]
I've double checked. So the only thing I can do would be use the location field. To determine if it's "Downloaded". But that would bring up another issue. And that is when a user archives a download. So all in all, with the current data model, it's not easy to implement sorry.

**Reply by IanKBrown** [reply]
OK well thanks for your efforts
